src/rp2_common/hardware_rtc/rtc.c

```diff
@@ -180,12 +234,20 @@ void rtc_set_alarm(datetime_t *t, rtc_callback_t user_callback) {
     irq_set_enabled(RTC_IRQ, true);
 
     rtc_enable_alarm();
+    return true;
 }
 
-void rtc_disable_alarm(void) {
-    // Disable matching and wait for it to stop being active
-    hw_clear_bits(&rtc_hw->irq_setup_0, RTC_IRQ_SETUP_0_MATCH_ENA_BITS);
-    while(rtc_hw->irq_setup_0 & RTC_IRQ_SETUP_0_MATCH_ACTIVE_BITS) {
-        tight_loop_contents();
-    }
+void rtc_delete_alarm(void)
+{
+    // first disable the alarm
+    rtc_disable_alarm();
+
+    // don't receive interrupts anymore
+    rtc_hw->inte = RTC_INTE_RESET;
+
+    // disable IRQ and remove handler
+    irq_set_enabled(RTC_IRQ, false);
+    irq_remove_handler(RTC_IRQ, _callback);
```

Comment:
Should this be `irq_remove_handler(RTC_IRQ, rtc_irq_handler)` ? :shrug: (IRQ-related code isn't my strong point!)

Reply:
nice, so I had in mind that I made too many changes with 1 commit ... and prompt forgot to fix this.

Reply:
done